staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -639,13 +628,9 @@ func (immediateRequest) Finish(execute func()) bool {
 // startRequest classifies and, if appropriate, enqueues the request.
 // Returns a nil Request if and only if the request is to be rejected.
 // The returned bool indicates whether the request is exempt from
-// limitation.  The hashFn and preQueueFn must not make any
-// synchronous calls on this controller.
-func (cfgCtl *configController) startRequest(ctx context.Context,
-	rd RequestDigest,
-	hashFn func(fsName string, distinguisherMethod *fctypesv1a1.FlowDistinguisherMethod) uint64,
-	preQueueFn func(),
-) (*fctypesv1a1.FlowSchema, *fctypesv1a1.PriorityLevelConfiguration, bool, fq.Request) {
+// limitation.  The startWaitingTime is when the request started
+// waiting in its queue, or `Time{}` if this did not happen.
```

Comment:
this could be ambiguous if someone starts an apiserver before midnight of Jan 1, 1970 :wink:

(not requesting a change)

Reply:
I figured the zero value is a natural way to say "didn't happen".